portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -5,9 +5,11 @@
   private String comment;
   private String posterName;
 
-  public Comment(String comment, String posterName) {
+  public Comment(String posterName, String comment) {
     this.comment = comment;
     this.posterName = posterName;
+    if(comment.trim().equals("")) this.comment = "\"Hey look at me I commented whitespace I'm so cool\"";
```

Comment:
make this a constant. Also I'd recommend using Optionals if you allow empty comments. 
Use something like
Optional<String> comment = Optional.ofNullable(str).filter(s -> !s.isEmpty());
and call this class with the optional. That would clean up the code in this constructor a lot.

Reply:
I am not familiar with Optionals, and would definitely like to talk about them in-depth at some point.